runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -332,27 +333,38 @@ String censorMessage(final String username, final String message)
 		boolean filtered = false;
 		for (Pattern pattern : filteredPatterns)
 		{
-			Matcher m = pattern.matcher(strippedMessage);
-
-			StringBuffer sb = new StringBuffer();
-
-			while (m.find())
+			// Match against both the original and normalized message so patterns with accented characters will still
+			// match correctly.
+			final Matcher[] matchers = {
+				pattern.matcher(strippedMessage),
+				pattern.matcher(Text.stripDiacritics(strippedMessage))
+			};
+			final List<MatchResult> matches = new ArrayList<>();
+
+			for (Matcher m : matchers)
 			{
-				switch (config.filterType())
+				while (m.find())
 				{
-					case CENSOR_WORDS:
-						m.appendReplacement(sb, StringUtils.repeat('*', m.group(0).length()));
-						filtered = true;
-						break;
-					case CENSOR_MESSAGE:
-						return CENSOR_MESSAGE;
-					case REMOVE_MESSAGE:
-						return null;
+					switch (config.filterType())
+					{
+						case CENSOR_WORDS:
+							matches.add(m.toMatchResult());
+							filtered = true;
+							break;
+						case CENSOR_MESSAGE:
+							return CENSOR_MESSAGE;
+						case REMOVE_MESSAGE:
+							return null;
+					}
 				}
 			}
-			m.appendTail(sb);
 
-			strippedMessage = sb.toString();
+			for (MatchResult match : matches)
+			{
+				strippedMessage = strippedMessage.substring(0, match.start())
```

Comment:
This looks like it is applying indexes from the match result of `Text.stripDiacritics(strippedMessage)` into `strippedMessage` which wouldn't always be correct